### PR TITLE
S4-56 View company information

### PR DESCRIPTION
**Pull request by faggc**
- Create controller, view, test stub
- Fixed unit tests for view company info controller and company service
- Fixes for address
- Added date filter
- Added mapper tests

### CODE REVIEW

File: src/filters/asCompanyStatusText.filter.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export interface CompanyStatusList {
+    [key: string]: string
+}
+
+export const companyStatusList: CompanyStatusList = {
+  'active': 'Active',
+  'dissolved': 'Dissolved',
```
Comment (mateuszc-kainos) [inline src/filters/asCompanyStatusText.filter.ts:7]: we don't need quotes on field names

File: src/views/view-company-information.njk [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    <div class="govuk-grid-column-two-thirds">
+      {% include 'components/errors/error-summary.njk' %}
+
+      <h1 class="govuk-heading-xl">View company information</h1>
+      {% if not company.canClose %}
+      <div id="cannot-close-error" class="govuk-inset-text">
+        <h2 class="govuk-heading-l">You cannot use this service</h2>
+        <p class="govuk-body">This could be because the company:</p>
+        <ul class="govuk-list govuk-list--bullet">
+          <li>is not active</li>
```
Comment (mateuszc-kainos) [inline src/views/view-company-information.njk:34]: '- has no active directors or members' bullet point missing

File: src/mappers/company/companyDetails.mapper.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return [company.registeredOfficeAddress.addressLineOne,
+      company.registeredOfficeAddress.addressLineTwo,
+      company.registeredOfficeAddress.careOf,
+      company.registeredOfficeAddress.country,
+      company.registeredOfficeAddress.locality,
+      company.registeredOfficeAddress.poBox,
+      company.registeredOfficeAddress.postalCode,
+      company.registeredOfficeAddress.premises,
+      company.registeredOfficeAddress.region]
+    .filter((el) => el != null)
```
Comment (mateuszc-kainos) [inline src/mappers/company/companyDetails.mapper.ts:33]: are the parentheses around 'el' needed?

File: src/views/view-company-information.njk (hunk 2 above)
Comment (faggc) [reply src/views/view-company-information.njk:34]: https://companieshouse.atlassian.net/browse/S4-56
see Jayti's comment, this is intentional

File: src/mappers/company/companyDetails.mapper.ts (hunk 3 above)
Comment (faggc) [reply src/mappers/company/companyDetails.mapper.ts:33]: good question

File: src/filters/asCompanyStatusText.filter.ts (hunk 1 above)
Comment (faggc) [reply src/filters/asCompanyStatusText.filter.ts]: Ive removed the quotes where possible

File: src/filters/asCompanyTypeText.filter.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+export interface CompanyTypeList {
+    [key: string]: string
+}
+
+export const companyTypeList: CompanyTypeList = {
+    'private-unlimited': 'Private unlimited company',
+    'ltd': 'Private limited company',
```
[the remark this answers is not in the document]
Comment (faggc) [reply src/filters/asCompanyTypeText.filter.ts]: Ive removed the quotes where possible

File: src/mappers/company/companyDetails.mapper.ts (hunk 3 above)
Comment (faggc) [reply src/mappers/company/companyDetails.mapper.ts]: it wasn't need, I have removed them

**Pull request merged**